### ISSUE TITLE
Unexpected error fetching Iec data - API changes(?)

### ISSUE THREAD

**Issue by tzagim** [GuyKh/py-iec-api#155]
### Describe the bug
```
Unexpected error fetching Iec data

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/iec/coordinator.py", line 350, in _async_update_data
    await self.api.load_jwt_token(
  File "/usr/local/lib/python3.12/site-packages/iec_api/iec_client.py", line 763, in load_jwt_token
    if await self.check_token():
       ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/iec_api/iec_client.py", line 796, in check_token
    await self.refresh_token()
  File "/usr/local/lib/python3.12/site-packages/iec_api/iec_client.py", line 804, in refresh_token
    self._token = await login.refresh_token(self._session, self._token)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/iec_api/login.py", line 206, in refresh_token
    response = await commons.send_post_request(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/iec_api/commons.py", line 183, in send_post_request
    parse_error_response(resp, json_resp)
  File "/usr/local/lib/python3.12/site-packages/iec_api/commons.py", line 103, in parse_error_response
    raise IECError(resp.status, resp.reason)
iec_api.models.exceptions.IECError: (Code 400): Bad Request
```

### To reproduce
For two days now, it is not possible to access the information through the addon.

### Additional context
Attempting to register again also encountered the same error.

**Comment by GuyKh** [comment]
IEC returns 400.
Please turn the logs to debug (see @ readme) and find the call and response for the calls to `https://iec-ext.okta.com/oauth2/default/v1/token` or `https://iec-ext.okta.com/api/v1/authn`). 
Let me know if you find something relevant there or else, just paste the entire log here or somewhere else.

**Comment by tzagim** [comment]
```
2024-11-10 09:15:21.602 DEBUG (MainThread) [custom_components.iec.coordinator] Finished fetching Iec data in 0.213 seconds (success: False)
2024-11-10 09:15:23.943 DEBUG (MainThread) [custom_components.iec.coordinator] Loading API token from config entry
2024-11-10 09:15:23.943 DEBUG (MainThread) [iec_api.iec_client] jwt.py token expired, refreshing token
2024-11-10 09:15:23.944 DEBUG (MainThread) [iec_api.commons] HTTP POST: https://iec-ext.okta.com/oauth2/default/v1/token
2024-11-10 09:15:23.944 DEBUG (MainThread) [iec_api.commons] HTTP POST: https://iec-ext.okta.com/oauth2/default/v1/token
2024-11-10 09:15:23.945 DEBUG (MainThread) [iec_api.commons] HTTP POST: https://iec-ext.okta.com/oauth2/default/v1/token
2024-11-10 09:15:23.945 DEBUG (MainThread) [iec_api.commons] HTTP POST: https://iec-ext.okta.com/oauth2/default/v1/token
2024-11-10 09:15:23.945 DEBUG (MainThread) [iec_api.commons] HTTP POST: https://iec-ext.okta.com/oauth2/default/v1/token
2024-11-10 09:15:23.945 DEBUG (MainThread) [iec_api.commons] HTTP POST: https://iec-ext.okta.com/oauth2/default/v1/token
2024-11-10 09:15:23.952 DEBUG (MainThread) [iec_api.commons] HTTP Content POST: b'client_id=XXXXXXXXXXXXXXXXX&redirect_uri=com.iecrn%3A%2F&refresh_token=XXXXXXXXXXXXXXXXX&grant_type=refresh_token&scope=openid+email+profile+offline_access'
2024-11-10 09:15:23.953 DEBUG (MainThread) [iec_api.commons] HTTP Content POST: b'client_id=XXXXXXXXXXXXXXXXX&redirect_uri=com.iecrn%3A%2F&refresh_token=XXXXXXXXXXXXXXXXX&grant_type=refresh_token&scope=openid+email+profile+offline_access'
2024-11-10 09:15:23.953 DEBUG (MainThread) [iec_api.commons] HTTP Content POST: b'client_id=XXXXXXXXXXXXXXXXX&redirect_uri=com.iecrn%3A%2F&refresh_token=XXXXXXXXXXXXXXXXX&grant_type=refresh_token&scope=openid+email+profile+offline_access'
2024-11-10 09:15:23.953 DEBUG (MainThread) [iec_api.commons] HTTP Content POST: b'client_id=XXXXXXXXXXXXXXXXX&redirect_uri=com.iecrn%3A%2F&refresh_token=XXXXXXXXXXXXXXXXX&grant_type=refresh_token&scope=openid+email+profile+offline_access'
2024-11-10 09:15:23.953 DEBUG (MainThread) [iec_api.commons] HTTP Content POST: b'client_id=XXXXXXXXXXXXXXXXX&redirect_uri=com.iecrn%3A%2F&refresh_token=XXXXXXXXXXXXXXXXX&grant_type=refresh_token&scope=openid+email+profile+offline_access'
2024-11-10 09:15:23.953 DEBUG (MainThread) [iec_api.commons] HTTP Content POST: b'client_id=XXXXXXXXXXXXXXXXX&redirect_uri=com.iecrn%3A%2F&refresh_token=XXXXXXXXXXXXXXXXX&grant_type=refresh_token&scope=openid+email+profile+offline_access'
2024-11-10 09:15:24.145 DEBUG (MainThread) [iec_api.commons] HTTP POST Response <400>: {"error":"invalid_grant","error_description":"The refresh token is invalid or expired."}
2024-11-10 09:15:24.145 DEBUG (MainThread) [iec_api.commons] HTTP POST Response <400>: {"error":"invalid_grant","error_description":"The refresh token is invalid or expired."}
2024-11-10 09:15:24.146 DEBUG (MainThread) [iec_api.commons] HTTP POST Response <400>: {"error":"invalid_grant","error_description":"The refresh token is invalid or expired."}
2024-11-10 09:15:24.146 DEBUG (MainThread) [iec_api.commons] HTTP POST Response <400>: {"error":"invalid_grant","error_description":"The refresh token is invalid or expired."}
2024-11-10 09:15:24.146 DEBUG (MainThread) [iec_api.commons] HTTP POST Response <400>: {"error":"invalid_grant","error_description":"The refresh token is invalid or expired."}
2024-11-10 09:15:24.146 DEBUG (MainThread) [iec_api.commons] HTTP POST Response <400>: {"error":"invalid_grant","error_description":"The refresh token is invalid or expired."}
2024-11-10 09:15:24.147 WARNING (MainThread) [iec_api.commons] Failed call: (Code 400): Bad Request
2024-11-10 09:15:24.147 ERROR (MainThread) [custom_components.iec.coordinator] Unexpected error fetching Iec data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/iec/coordinator.py", line 350, in _async_update_data
    await self.api.load_jwt_token(
  File "/usr/local/lib/python3.12/site-packages/iec_api/iec_client.py", line 763, in load_jwt_token
    if await self.check_token():
       ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/iec_api/iec_client.py", line 796, in check_token
    await self.refresh_token()
  File "/usr/local/lib/python3.12/site-packages/iec_api/iec_client.py", line 804, in refresh_token
    self._token = await login.refresh_token(self._session, self._token)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/iec_api/login.py", line 206, in refresh_token
    response = await commons.send_post_request(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/iec_api/commons.py", line 183, in send_post_request
    parse_error_response(resp, json_resp)
  File "/usr/local/lib/python3.12/site-packages/iec_api/commons.py", line 103, in parse_error_response
    raise IECError(resp.status, resp.reason)
iec_api.models.exceptions.IECError: (Code 400): Bad Request
2024-11-10 09:15:24.155 DEBUG (MainThread) [custom_components.iec.coordinator] Finished fetching Iec data in 0.212 seconds (success: False)
```

**Comment by GuyKh** [comment]
Try to remove the integration and add it again - it seems that the refresh token is invalid.

**Comment by GuyKh** [comment]
Note that you've opened this ticket in the python package - what you're describing is related to - https://github.com/GuyKh/iec-custom-component

**Comment by tzagim** [comment]
After uninstall > reboot > install > reboot.
I try to register and I get the following error "Unknown error occurred"

I opened it here because I thought they were changing the API configuration

**Comment by tzagim** [comment]
It could be related to the existence of more than one meter in the account?
```
2024-11-10 09:25:29.490 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "<string>", line 18, in __mashumaro_from_dict__
  File "<string>", line 22, in __mashumaro_from_dict__
mashumaro.exceptions.MissingField: Field "last_name" of type str is missing in CustomerAccount instance
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 477, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 559, in _handle
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 92, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 210, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 26, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 86, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 242, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 32, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/http.py", line 73, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/decorators.py", line 81, in with_admin
    return await func(self, request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 222, in post
    return await super().post(request, flow_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 74, in wrapper
    return await method(view, request, data, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 122, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 367, in async_configure
    result = await self._async_configure(flow_id, user_input)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 414, in _async_configure
    result = await self._async_handle_step(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 517, in _async_handle_step
    result: _FlowResultT = await getattr(flow, method)(user_input)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/iec/config_flow.py", line 130, in async_step_mfa
    customer = await client.get_customer()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/iec_api/iec_client.py", line 101, in get_customer
    customer = await data.get_customer(self._session, self._token)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/iec_api/data.py", line 131, in get_customer
    return Customer.from_dict(response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 20, in __mashumaro_from_dict__
mashumaro.exceptions.InvalidFieldValue: Field "accounts" of type list[CustomerAccount] in Customer has invalid value [{'mainContractId':........
```

**Comment by GuyKh** [comment]
As you can see, it's a different error;
It's possible, but I need to see the full error + response to understand what invalid value

**Comment by tzagim** [comment]
Indeed, it appears that a non-private account has no lastName.

**Comment by GuyKh** [comment]
@tzagim - I'd love to see the entire response to see that nothing else is missing

**Comment by tzagim** [comment]
@GuyKh 
This is the only line I left out:
```
mashumaro.exceptions.InvalidFieldValue: Field "accounts" of type list[CustomerAccount] in Customer has invalid value
[
{'mainContractId': '1111111111111', 'mainContractIdType': 1, 'companyId': '111111111', 'name': 'השם שלי', 'lastName': 'המשפחה', 'bpNumber': '01010101010', 'bpType': 1, 'isActiveAccount': True, 'customerRole': 0, 'accountType': 1},
{'mainContractId': '11111111111111111', 'mainContractIdType': 1, 'companyId': '111111111', 'name': 'שם חשבון 2', 'bpNumber': '123321123', 'bpType': 3, 'isActiveAccount': True, 'customerRole': 1, 'accountType': 3}
]
```

**Comment by GuyKh** [comment]
@tzagim - Looks like OK. Merging and updating `iec-custom-component`

**Comment by GuyKh** [comment]
Fixed in `iec-custom-component` `0.41.1`
